connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/SparkConnectClientSuite.scala

```diff
@@ -123,6 +122,28 @@ class SparkConnectClientSuite extends ConnectFunSuite with BeforeAndAfterEach {
     assert(df.plan === service.getAndClearLatestInputPlan())
   }
 
+  test("CustomInterceptor") {
+    startDummyServer(0)
+    client = SparkConnectClient
+      .builder()
+      .connectionString(s"sc://localhost:${server.getPort}")
+      .interceptor(new ClientInterceptor {
+        override def interceptCall[ReqT, RespT](
+                                                 methodDescriptor: MethodDescriptor[ReqT, RespT],
+                                                 callOptions: CallOptions,
+                                                 channel: Channel): ClientCall[ReqT, RespT] = {
+          throw new RuntimeException("Blocked")
+        }
+      })
+      .build()
+
+    val session = SparkSession.builder().client(client).create()
```

Comment:
The `client()` method is `private[sql]` for anyone providing any kinds of wrappers on top of the SparkSession for custom auth integrations it's not possible to use this method.

Reply:
The proposed usage is rather

`SparkSession.builder().interceptor(my interceptor).create()`

Perhaps I should add a test doing this exact line nearby

Reply:
Now there is an external api test as well (in SparkSessionSuite)

Reply:
Now there is an external api test as well (in SparkSessionSuite)